app/hooks/useWalletTokens.ts

```diff
@@ -163,6 +164,24 @@ export const useWalletTokens = (walletsState: ConnectedWalletsState) => {
     };
   };
 
+  const getCosmosBalance = async (walletAddress: string) => {
+    try {
+      const client = await SigningStargateClient.connect(
+        "https://rpc.cosmos.directory/cosmoshub",
+      );
+
+      const balance = await client.getBalance(walletAddress, "uatom");
+
+      return {
+        balance: Number(balance.amount) / 1e6, // Convert from uatom to ATOM
+        formattedBalance: formatNumber(Number(balance.amount) / 1e6, 6),
```

Comment:
I think it would be better to use Xchain's functions 
- baseAmount
- assetAmount

These work with BigNumber and will avoid accuracy problems.
This code using these utilities would be:

`baseToAsset(baseAmount(balance.amount, 6)).amount().toNumber()`

Same for the next line